### PR TITLE
PAYM-2893 Refactor event envelope

### PR DESCRIPTION
**Pull request by dabeat**
#### :tophat: Problem
Event structure is defined in AWS adapter, out of the business logic modules, making really complicated to do JSON schema validation.

#### :pushpin: Solution
**1st step**: Remove event structure transformations from adapter and locate it in `Event` (business) module, so that event JSON schema validation can be properly done.

#### :ghost: GIF
giphy

### CODE REVIEW

File: lib/events/adapters/aws_sns.ex [hunk 1]
```diff
@@ -5,20 +5,16 @@ defmodule ElixirTools.Events.Adapters.AwsSns do
...
            {:aws_module, module}
            | {:sns_module, module}
-           | {:uuid_module, module}
            | {:group, String.t()}
            | {:topic, String.t()}
-           | {:timex_module, module}
 
   @impl true
-  @spec publish(Event.t(), [publish_opt]) :: :ok | {:error, term}
+  @spec publish(map, [publish_opt]) :: :ok | {:error, term}
```
Comment (dabeat) [inline lib/events/adapters/aws_sns.ex:17]: My suggestion is not to create two schema modules, which would make over complicated the event solution. 

Current Event schema should be migrated to the event structure after having added the envelope. This is transparent for clients as event is never directly manipulated, but directly piped to `Event.publish/2` method.

When done, this spec could be moved back to `@spec publish(Event.t(), [publish_opt]) :: :ok | {:error, term}`

File: lib/events/adapters/aws_sns.ex [hunk 2]
```diff
@@ -5,20 +5,16 @@ defmodule ElixirTools.Events.Adapters.AwsSns do
...
 
-  alias ElixirTools.Events.Event
-
   @behaviour ElixirTools.Events.Publisher
 
   @typep publish_opt ::
            {:aws_module, module}
            | {:sns_module, module}
-           | {:uuid_module, module}
            | {:group, String.t()}
```
Comment (Kitton) [inline lib/events/adapters/aws_sns.ex:13]: not needed anymore

File: lib/events/adapters/aws_sns.ex (hunk 1 above)
Comment (Kitton) [reply lib/events/adapters/aws_sns.ex:17]: :thinking: in this case how specs of `EventHandler.create | publish` will looks like?

File: test/events/event_test.exs [hunk 3]
```diff
@@ -12,7 +12,31 @@ defmodule ElixirTools.Events.EventTest do
...
+      :ok
+    end
+  end
+
+  defmodule FakeUuid do
+    use ElixirTools.ContractImpl, module: UUID
+
+    @impl true
+    def uuid5(uuid_seed_1, uuid_seed_2) do
+      send(self(), [:uuid5, uuid_seed_1, uuid_seed_2])
```
Comment (Kitton) [inline test/events/event_test.exs:27]: never asserted

File: lib/events/adapters/aws_sns.ex (hunk 1 above)
Comment (dabeat) [reply lib/events/adapters/aws_sns.ex:16]: They should look exactly the same:

```elixir
@spec create(event_name, payload, event_id_seed, [create_optional]) :: Event.t()
```

as the `Event.t()` structure should be modified to match with an event such us:

```elixir
%Event{
        action: "TEST_EVENT",
        group: "EVENT_GROUP",
        id: "55989ea4-d947-4f16-b8c8-e0e888facff4",
        occurred_at: "2015-06-29T05:55:55Z",
        payload: %{},
        version: "1.0.0"
      }
```

For th external world, in terms of specs should be transparent. The only difference would be the structure of the event, The event will be, after created, the final event (with the "envelope").

JSON schema validation could even be done here, before publish call. This way, JSON schema validation would be done with the rest of validations that are currently being done.

**Pull request merged**